packages/snap-client/src/Client/apis/Abstract.test.ts

```diff
@@ -0,0 +1,265 @@
+			queryParamsStringify: queryParamMockfn,
+		};
+
+		let api;
+		// end set up
+
+		expect(() => {
+			api = new API(new ApiConfiguration(config));
+		}).not.toThrow();
+
+		//can use query params stringify
+		expect(api.configuration.queryParamsStringify(['string', 'thing'])).toBe('here');
+		expect(queryParamMockfn).toHaveBeenCalled();
+	});
+
+	it('can use createFetchParams', async () => {
+		let config: ApiConfigurationParameters = {
+			origin: 'https://searchspring.com',
+			fetchApi: global.window.fetch,
+			headers: customHeaders,
+		};
+
+		let api;
+		// end set up
+
+		expect(() => {
+			api = new API(new ApiConfiguration(config));
+		}).not.toThrow();
+
+		const body = { siteId: '8uyt2m' };
+		const context = {
+			path: '/api/v1/autocomplete',
+			method: 'POST',
+			headers: config.headers,
+			body: body,
+		};
+
+		//can use createFetchParams and get back expected values
+		const fetchParams = api.createFetchParams(context);
+		expect(fetchParams).toBeDefined();
+		expect(fetchParams.url).toBe(`${config.origin}/api/v1/autocomplete`);
+		expect(fetchParams.init.body).toBe(JSON.stringify(body));
+		expect(fetchParams.init.headers).toStrictEqual(customHeaders);
+		expect(fetchParams.init.method).toBe('POST');
+
+		//create fetch params requires a site id
+		const badContext = { ...context, body: {} };
+		expect(() => api.createFetchParams(badContext)).toThrowError(`Request failed. Missing "siteId" parameter.`);
+
+		let config2: ApiConfigurationParameters = {
+			origin: 'https://searchspring.com',
+			fetchApi: global.window.fetch,
+			headers: customHeaders,
+			maxRetry: 2,
+			cache: CustomCacheConfig,
+		};
+		let api2 = new API(new ApiConfiguration(config2));
+
+		const contextWithQuery = { ...context, query: { key: 'value' } };
+		// @ts-ignore
+		const params = api2.createFetchParams(contextWithQuery);
+		expect(params.url).toBe('https://searchspring.com/api/v1/autocomplete?key=value');
+	});
+
+	it('can use cacheKey', async () => {
+		let config: ApiConfigurationParameters = {
+			origin: 'https://searchspring.com',
+			fetchApi: global.window.fetch,
+			headers: customHeaders,
+		};
+
+		let api;
+		expect(() => {
+			api = new API(new ApiConfiguration(config));
+		}).not.toThrow();
+
+		const body = { siteId: '8uyt2m' };
+		const context = {
+			path: '/api/v1/autocomplete',
+			method: 'POST',
+			headers: config.headers,
+			body: body,
+		};
+
+		//can use the cacheKey
+		const cacheKey = '/api/v1/autocomplete' + JSON.stringify(body);
+
+		let request;
+		request = await api.request(context, cacheKey);
+
+		expect(fetchfn).toHaveBeenCalled();
+
+		fetchfn.mockClear();
+
+		//handles 200s
+		const fetchfn200 = jest
+			.spyOn(global.window, 'fetch')
+			.mockImplementation(() => Promise.resolve({ status: 200, json: () => Promise.resolve({ found: true }) } as unknown as Response));
+
+		const setCache = jest.fn();
+		const getCache = jest.fn(() => Promise.resolve({ foundInCache: true }));
+
+		api.cache.set = setCache;
```

Comment:
Overriding class methods under tests is not ideal. Can we use and/or spy on the actual cache set/get methods?